samples/Balosar/Balosar.Server/Worker.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Balosar.Server.Data;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenIddict.Abstractions;
+using OpenIddict.Core;
+using OpenIddict.EntityFrameworkCore.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static OpenIddict.Abstractions.OpenIddictConstants;
+
+namespace Balosar.Server
+{
+    public class Worker : IHostedService
+    {
+        private readonly IServiceProvider serviceProvider;
+
+        public Worker(IServiceProvider serviceProvider) =>
+            this.serviceProvider = serviceProvider;
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            using var scope = serviceProvider.CreateScope();
+
+            var context = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+            await context.Database.EnsureCreatedAsync();
+            
+            var manager = scope.ServiceProvider.GetRequiredService<OpenIddictApplicationManager<OpenIddictEntityFrameworkCoreApplication>>();
+
+            if (await manager.FindByClientIdAsync("balosar-blazor-client") is null)
+            {
+                await manager.CreateAsync(new OpenIddictApplicationDescriptor
+                {
+                    ClientId = "balosar-blazor-client",
+                    ConsentType = ConsentTypes.Explicit,
+                    DisplayName = "Blazor client application",
+                    Type = ClientTypes.Public,
+                    PostLogoutRedirectUris =
+                    {
+                        new Uri("http://localhost:51818/authentication/logout-callback"),
+                        new Uri("https://localhost:44310/authentication/logout-callback")
+                    },
+                    RedirectUris =
+                    {
+                        new Uri("http://localhost:51818/authentication/login-callback"),
+                        new Uri("https://localhost:44310/authentication/login-callback")
+                    },
+                    Permissions =
+                    {
+                        Permissions.Endpoints.Authorization,
+                        Permissions.Endpoints.Logout,
+                        Permissions.Endpoints.Token,
+                        Permissions.GrantTypes.AuthorizationCode,
+                        Permissions.GrantTypes.RefreshToken,
+                        Permissions.Scopes.Email,
+                        Permissions.Scopes.Profile,
+                        Permissions.Scopes.Roles
+                    },
+                    Requirements =
+                    {
+                        Requirements.Features.ProofKeyForCodeExchange
+                    }
+                });
+            }
+
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
```

Comment:
Can we return a completed task instead of throwing an exception?

```csharp
public Task StopAsync(CancellationToken cancellationToken) => Task.CompletedTask;
```